File: longWords.js
let list = [ 'word', 'localization', 'internationalization', 'pneumonoultramicroscopicsilicovolcanoconiosis' ];

const checkString = list.map( ( m ) =>
{
    // console.log(m)
    // console.log(m.length)
    // console.log( m[0])
    // console.log(m[m.length - 1])
    if ( m.length <= 10 )
    {
        return m
    }
    else
    {
      return `${m[1]}${m.length-2}${m[m.length-1]}`  
    }
} );

console.log(checkString)

File: longWords.cpp
#include <stdio.h>
#include <string.h>

void checkString(char *str)
{
    int len = strlen(str);
    if (len > 10)
    {
        printf("%c%d%c\n", str[0], len - 2, str[len - 1]);
    }
    else
    {
        printf("%s\n", str);
    }
};

int main()
{
    int n;
    scanf("%d", &n);

    for (int i = 0; i < n; i++)
    {
        char word[100];
        scanf("%s", word);
        // printf("%s\n", word);
        checkString(word);
    };

    return 0;
};